### DISCUSSION TITLE
Total N00b question about entities (python)

### DISCUSSION THREAD

**Discussion by skintflickz** [explosion/spaCy#11712] [Help: Coding & Implementations] [answered]
I'm sure anyone on here can answer this question but I'm new to python so could really do with a bit of help :)

Basically I want to extract and remove all labels except the entity

example from PERSON i just want to add all the names (nothing else) to list so that I can remove any duplicates.

I can add the entities to list but they still have the labels attached and I cant then remove the duplicates

example of my code that does look like it works but dose not.

```python
people = []
places = []
company = []

def spacyents():
    spacy.prefer_gpu()
    nlp = spacy.load("en_core_web_lg")
    doc = nlp(testdata())
   
    for ent in doc.ents:        
        if ent.label_ == "PERSON":
            people.append(ent)
        if ent.label_ == "ORG":
            company.append(ent)
        if ent.label_ == "GPE":
            places.append(ent)`

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @skintflickz , perhaps you're looking for [`ent.text`](https://spacy.io/api/span#attributes)? It gives you a `str` that you can use to remove duplicates (e.g., `set(people)`). 

```python
for ent in doc.ents:
    if ent.label_ == "PERSON":
        people.append(ent.text)
```

**Comment by skintflickz** [comment]
sorted

https://www.reddit.com/r/learnpython/comments/yexien/confused_about_type_of_array_or_list_i_generate/
